### PR TITLE
ClickHouse Cloud promotion

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add some good advertisement for the Cloud.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit ada1b3b45381f75f0d4c32764c736fb4466fe3d3 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57638/ada1b3b45381f75f0d4c32764c736fb4466fe3d3/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/7134127065">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57638/ada1b3b45381f75f0d4c32764c736fb4466fe3d3/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Core/Settings.h [hunk 1]
```diff
@@ -526,6 +526,8 @@ class IColumn;
     M(Int64, max_partitions_to_read, -1, "Limit the max number of partitions that can be accessed in one query. <= 0 means unlimited.", 0) \
     M(Bool, check_query_single_value_result, true, "Return check query result as single 1/0 value", 0) \
     M(Bool, allow_drop_detached, false, "Allow ALTER TABLE ... DROP DETACHED PART[ITION] ... queries", 0) \
+    M(UInt64, max_table_size_to_drop, 0, "Only available in ClickHouse Cloud", 0) \
```
Comment (alexey-milovidov) [inline src/Core/Settings.h:529]: The description should be full. It should be identical in the public and the private repositories.

File: src/Storages/MergeTree/MergeTreeSettings.h [hunk 2]
```diff
@@ -170,13 +183,18 @@ struct Settings;
     M(UInt64, zero_copy_merge_mutation_min_parts_size_sleep_before_lock, 1ULL * 1024 * 1024 * 1024, "If zero copy replication is enabled sleep random amount of time before trying to lock depending on parts size for merge or mutation", 0) \
     M(Bool, allow_floating_point_partition_key, false, "Allow floating point as partition key", 0) \
     M(UInt64, sleep_before_loading_outdated_parts_ms, 0, "For testing. Do not change it.", 0) \
+    M(Bool, always_use_copy_instead_of_hardlinks, false, "Only available in ClickHouse Cloud", 0) \
```
Comment (alexey-milovidov) [inline src/Storages/MergeTree/MergeTreeSettings.h:186]: Let's port the hacks and kludges for zero-copy replication to the public.

File: src/Core/Settings.h (hunk 1 above)
Comment (javisantana) [reply src/Core/Settings.h:529]: not sure if the plan is to keep this feature to the internal fork but I don't think it should be in the open-source version if there is no such feature

Comment (alexey-milovidov) [reply src/Core/Settings.h:529]: This will be available in open-source.

**Comment by nikitamikhaylov** [comment]
`test_storage_hdfs/test.py::test_write_table` - https://github.com/ClickHouse/ClickHouse/issues/57819
`test_storage_azure_blob_storage/test_cluster.py::test_unset_skip_unavailable_shards` - https://github.com/ClickHouse/ClickHouse/issues/57889
`test_postgresql_replica_database_engine_1/test.py::test_many_concurrent_queries ` - Timeout for `SYSTEM SYNC REPLICA`
`test_replicated_merge_tree_s3_zero_copy/test.py::test_drop_table` - Timeout for `SYSTEM SYNC REPLICA`

File: docker/test/upgrade/run.sh [hunk 3]
```diff
@@ -77,6 +77,7 @@ remove_keeper_config "create_if_not_exists" "[01]"
 # it contains some new settings, but we can safely remove it
 rm /etc/clickhouse-server/config.d/merge_tree.xml
 rm /etc/clickhouse-server/config.d/enable_wait_for_shutdown_replicated_tables.xml
+rm /etc/clickhouse-server/config.d/zero_copy_destructive_operations.xml
```
Comment (tavplubix) [inline docker/test/upgrade/run.sh:80]: We need the same change for Tests Bugfix Validate Check:
https://s3.amazonaws.com/clickhouse-test-reports/57906/e4a831be17f85e345c5c9cd8f00bd22b8e759692/bugfix_validate_check.html
https://s3.amazonaws.com/clickhouse-test-reports/57906/e4a831be17f85e345c5c9cd8f00bd22b8e759692/stateless_tests_bugfix_validate_check.html
https://s3.amazonaws.com/clickhouse-test-reports/57906/e4a831be17f85e345c5c9cd8f00bd22b8e759692/stateless_tests_bugfix_validate_check/clickhouse-server.err.log

Comment (tavplubix) [reply docker/test/upgrade/run.sh:80]: https://github.com/ClickHouse/ClickHouse/pull/58078

**Pull request merged**